nucypher/cli/actions/confirm.py

```diff
@@ -115,3 +121,31 @@ def confirm_destroy_configuration(config: CharacterConfiguration) -> bool:
                                                 database=database)
     click.confirm(confirmation, abort=True)
     return True
+
+
+def verify_upgrade_details(contract_name: str,
+                           blockchain: BlockchainDeployerInterface,
+                           local_registry: LocalContractRegistry,
+                           source_registry: InMemoryContractRegistry,
+                           deployer: Type[BaseContractDeployer],
+                           ) -> None:
+    """
+    Compares the versions of two 'implementation' contracts using a local and source registry.
+    """
+    old_contract: VersionedContract = blockchain.get_contract_by_name(
+        registry=source_registry,
+        contract_name=deployer.contract_name,
```

Comment:
You pass `contract_name` as a parameter to this function - should we just remove it since it is obtained from the deployer?

Reply:
Hmm - not sure I understand your question.

Reply:
`contract_name` is a parameter, and so is `deployer`.

Within this function we use both `deployer.contract_name` (except on L149, which I assume can use `deployer.contract_name` as well). Can we simply use `deployer.contract_name` and no longer have `contract_name` as a parameter?

Reply:
Oh, I see.  That implies a deeper change to an existing intermediate API. It's not always the case that a `deployer` is in context when calling `blockchain.get_contract_by_name.`

Reply:
Oh - you meant for `verify_upgrade_details` - Got it.  Yes I'll make that change.

Reply:
👍